### PR TITLE
Include section on validating entitlements JWT

### PR DESCRIPTION
**Pull request by ed-bird**
@ratana  - FYI

### CODE REVIEW

File: docs/entitlements-flow.md [hunk 1]
```diff
@@ -87,6 +87,24 @@ subscriptions.setOnEntitlementsResponse(function(entitlementsPromise) {
   });
 })
 ```
+## Validate entitlements
```
Comment (dvoytenko) [inline docs/entitlements-flow.md]: nit: +1 empty line

File: docs/entitlements-flow.md [hunk 2]
```diff
@@ -87,6 +87,24 @@ subscriptions.setOnEntitlementsResponse(function(entitlementsPromise) {
...
+To verify the entitlements from Google are valid and authorised, check the validity of the JWT representation of the entitlements.
+
+The following registered claims are used:
+
+| Claim | Value | Description |
+| ----- | ----- | ----------- |
+| `exp` | [NumericDate](https://tools.ietf.org/html/rfc7519#page-6) (e.g. `1543596936` ) | When the token expires |
```
Comment (dvoytenko) [inline docs/entitlements-flow.md]: Use this URL: https://tools.ietf.org/html/rfc7519#section-4.1.4

File: docs/entitlements-flow.md [hunk 3]
```diff
@@ -87,6 +87,24 @@ subscriptions.setOnEntitlementsResponse(function(entitlementsPromise) {
...
+| `iat` | [NumericDate](https://tools.ietf.org/html/rfc7519#page-6) (e.g. `1543595136` ) | When the token was issued |
```
Comment (dvoytenko) [inline docs/entitlements-flow.md]: Use https://tools.ietf.org/html/rfc7519#section-4.1.6

File: docs/entitlements-flow.md [hunk 4]
```diff
@@ -87,6 +87,24 @@ subscriptions.setOnEntitlementsResponse(function(entitlementsPromise) {
...
+| `iss` | `subscribewithgoogle@system.gserviceaccount.com` | Issuer of the token |
+| `aud` | URL (e.g. `https://example.com` ) | Audience of the token which will be the publisher's URL |
```
Comment (dvoytenko) [inline docs/entitlements-flow.md]: It'd be publisher's origin, not URL.

File: docs/entitlements-flow.md [hunk 5]
```diff
@@ -87,6 +87,24 @@ subscriptions.setOnEntitlementsResponse(function(entitlementsPromise) {
...
+
+To validate the signature, use the keys found here:
+https://www.googleapis.com/robot/v1/metadata/x509/subscribewithgoogle@system.gserviceaccount.com
```
Comment (dvoytenko) [inline docs/entitlements-flow.md]: Please use `[]()` link markup. And let's try to combine these two into a single sentence. E.g. "[x509 format] and [JWK format]".

File: docs/entitlements-flow.md (hunk 1 above)
Comment (ed-bird) [reply docs/entitlements-flow.md]: Done.

File: docs/entitlements-flow.md (hunk 2 above)
Comment (ed-bird) [reply docs/entitlements-flow.md]: Kept the link to the NumericDate specification but added a link directly to the spec for the different claims.

File: docs/entitlements-flow.md (hunk 3 above)
Comment (ed-bird) [reply docs/entitlements-flow.md]: Kept the link to the NumericDate specification but added a link directly to the spec for the different claims.

File: docs/entitlements-flow.md (hunk 4 above)
Comment (ed-bird) [reply docs/entitlements-flow.md]: Done.

File: docs/entitlements-flow.md (hunk 5 above)
Comment (ed-bird) [reply docs/entitlements-flow.md]: Done.

**Comment by ed-bird** [comment]
Made some changes, let me know if there is anything else

**Pull request merged**